src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,82 @@
+package duke;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.ListIterator;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the collection of Tasks set by a user of Duke, stored within a ArrayList.
+ */
+public class TaskList {
+    private final List<Task> tasks;
+
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Updates a Task at a given index in the collection as complete.
+     * @return Task updated as complete.
+     */
+    protected Task markTask(int index) {
+        Task performedTask = tasks.get(index).performTask();
+        tasks.set(index, performedTask);
+        return performedTask;
+    }
+
+    /**
+     * Updates a Task at a given index in the collection as incomplete.
+     * @return Task updated as incomplete.
+     */
+    protected Task unmarkTask(int index) {
+        Task undidTask = tasks.get(index).undoTask();
```

Comment:
`undidTask` doesn't seem to be standard English. Perhaps you could rename it to `incompleteTask`.
```suggestion
        Task incompleteTask = tasks.get(index).undoTask();
```